submissions/usides/friends-app/js/main.js

```diff
@@ -0,0 +1,149 @@
+const HUMANS_LINK =
+  'https://randomuser.me/api/?nat=us&results=36&inc=gender,name,picture,dob&seed=06b64d62bc01d624a';
+
+const searchBtn = document.getElementById('search-btn');
+const rightPane = document.getElementById('right-pane');
+const cardsHolder = document.getElementById('cards-holder');
+const optionGroup = document.getElementById('option-group');
+const sortButtons = document.querySelectorAll('button[data-sort]');
+const filterButtons = document.querySelectorAll('button[data-filter]');
+const nameInput = document.getElementById('name-input');
+const errorNotice = document.getElementById('error-notice');
+const loader = document.getElementById('loader');
+
+let humansOriginState;
+
+const currentState = {
+  humans: undefined,
+  sort: 'a-z',
+  filter: 'none',
+  searchValue: '',
+};
+
+const handleSearcBtn = function () {
+  rightPane.classList.toggle('visible');
+};
+
+const handleInput = function () {
+  currentState.searchValue = this.value.toLowerCase();
+  filterHumans(currentState.filter);
+  sortHumans(currentState.sort);
+  updateCards(currentState.humans);
+};
+
+const handleOptionClick = function (e) {
+  const button = e.target.closest('button');
+  if (!button) return;
+  if (button.dataset.sort) {
+    sortButtons.forEach((button) => button.classList.remove('active'));
+    button.classList.add('active');
+    sortHumans(button.dataset.sort);
+  }
+  if (button.dataset.filter) {
+    if (currentState.filter === button.dataset.filter) {
+      button.classList.toggle('active');
+      filterHumans('none');
+    } else {
+      filterButtons.forEach((button) => button.classList.remove('active'));
+      button.classList.add('active');
+      filterHumans(button.dataset.filter);
+    }
+    sortHumans(currentState.sort);
+  }
+  updateCards(currentState.humans);
+};
+
+const filterHumans = function (filterOption) {
+  const searchedName = humansOriginState.filter((elem) =>
```

Comment:
Consider renaming this variable because now it's confusing a little bit.
Maybe `humansFilteredByName` would be better.

Reply:
Agree.